src/clients/meta/MetaClient.cpp

```diff
@@ -743,7 +743,8 @@ void MetaClient::getResponse(Request req,
                 // succeeded
                 pro.setValue(respGen(std::move(resp)));
                 return;
-              } else if (code == nebula::cpp2::ErrorCode::E_LEADER_CHANGED) {
+              } else if (code == nebula::cpp2::ErrorCode::E_LEADER_CHANGED ||
+                         code == nebula::cpp2::ErrorCode::E_MACHINE_NOT_FOUND) {
```

Comment:
If so, why not just handle only return `E_MACHINE_NOT_FOUND` when the key not in `ActiveHostsMan::machineRegisted`...

Reply:
when the active hosts manager not found the hosts return `E_MACHINE_NOT_FOUND` and set leader message

Reply:
I mean only leader could return `E_MACHINE_NOT_FOUND`, follower will return LEADER_CHANGED, so we don't need to have to add code in `MetaClient`